tests/suites/main_test.function

```diff
@@ -395,7 +405,50 @@ int main(int argc, const char *argv[])
             if( unmet_dep_count == 0 )
             {
                 test_errors = 0;
+
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+                /* Suppress all output from the library unless we're verbose
+                 * mode
+                 */
+                if( !option_verbose )
+                {
```

Comment:
I would prefer if the content of this `if` statement and the one after the call to `dispatch` were placed inside functions like `redirect_io()` or something like this.  The code for the main is already quite long... However, this is just a suggestion, not a real problem.